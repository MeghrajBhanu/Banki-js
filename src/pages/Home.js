import React from "react";
import { Link } from "react-router-dom";
import homeimage from "../assets/homeimage.jpg";
const Home = () => {
  return (
    <>
      <div className="container-fluid h-full " style={{ height: "100vh" }}>
        <img
          className=" container-fluid align-items-center"
          src={homeimage}
          style={{
            height: "100%",
            imageRendering: "-webkit-optimize-contrast",
          }}
          alt="managing people"
        ></img>
        <div className="h-full row w-100 align-items-center justify-content-center">
          <div className="col">
            <div className="row align-items-center">
              <div className="col-12 h3 text-center text-black mt-4">
                Welcome
              </div>

              <span className="lead-text text-center p-3 h5">
                We are a services group focused on providing customers{" "}
              </span>
              <br></br>
              <span className="lead-text text-center p-2 h5">
                with all there account details associated with different banks
                at a single point
              </span>
              <h6 className="text-center p-1">
                Don't have a account yet? <Link to="/register">Register</Link>
              </h6>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};
export default Home;