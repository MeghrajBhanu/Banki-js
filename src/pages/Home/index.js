import React from "react";
import { Link } from "react-router-dom";
import home from "../../assets/home.jpg";
import classes from "./index.module.css";
import "aos/dist/aos.css";

/**
 * Home component is default page when user is not logged in
 *
 * @returns {React.ReactElement}
 */
const Home = () => {
  return (
    <>
      <div className={classes["first_container"]}>
        <img src={home} className={classes["image_flex"]} alt="landing"></img>

        <div className={classes["content"]}>
          <p className="lead-text text-center p-2 h5 mt-5">
            We are a services group focused on providing you
            <br />
            <br />
            with all your account details associated with different
            <br />
            <br />
            banks at a single point
          </p>
          <h6 className="text-center p-1">
            Don't have a account yet? <Link to="/register">Register</Link>
          </h6>
        </div>
      </div>

      <section
        id="services"
        className={`services ${classes["section-bg"]} mt-4`}
      >
        <div className="container aos-init aos-animate" data-aos="fade-up">
          <div className={classes["section-header"]}>
            <h2 style={{ padding: "20px" }}>About Us</h2>
            <p>12Lakh+ customers use our website on daily basis.</p>
          </div>

          <div className="row gy-4">
            <div
              className="col-lg-4 col-md-6 aos-init aos-animate"
              data-aos="fade-up"
              data-aos-delay="100"
            >
              <div className={`${classes["service-item"]}  position-relative`}>
                <div className={classes["icon"]}>
                  <i className="fa-solid fa-mountain-city"></i>
                </div>
                <h3>Centralizes Access</h3>
                <p className="h6">
                  We provide centralized access to all your bank accounts
                </p>
                
              </div>
            </div>

            <div
              className="col-lg-4 col-md-6 aos-init aos-animate"
              data-aos="fade-up"
              data-aos-delay="200"
            >
              <div className={`${classes["service-item"]}  position-relative`}>
                <div className={classes["icon"]}>
                  <i className="fa-solid fa-arrow-up-from-ground-water"></i>
                </div>
                <h3>Data Security</h3>
                <p className="h6">
                  Keeping your private data safe is our priority.
                </p>
                <a href="service-details.html" className="readmore stretched-link">
                  Learn more <i className="bi bi-arrow-right"></i>
                </a>
              </div>
            </div>

            <div
              className="col-lg-4 col-md-6 aos-init aos-animate"
              data-aos="fade-up"
              data-aos-delay="300"
            >
              <div className={`${classes["service-item"]}  position-relative`}>
                <div className={classes["icon"]}>
                  <i className="fa-solid fa-compass-drafting"></i>
                </div>
                <h3>Leaders of Market</h3>
                <p className="h6">
                  We are a leading centralized service provider.
                </p>
                <a href="service-details.html" className="readmore stretched-link">
                  Learn more <i className="bi bi-arrow-right"></i>
                </a>
              </div>
            </div>

            <div
              className="col-lg-4 col-md-6 aos-init aos-animate"
              data-aos="fade-up"
              data-aos-delay="400"
            >
              <div className={`${classes["service-item"]}  position-relative`}>
                <div className={classes["icon"]}>
                  <i className="fa-solid fa-trowel-bricks"></i>
                </div>
                <h3>User Friendly</h3>
                <p className="h6">
                  Our User Experience make us apart from all otherservice providers.
                </p>
                
              </div>
            </div>

            <div
              className="col-lg-4 col-md-6 aos-init aos-animate"
              data-aos="fade-up"
              data-aos-delay="500"
            >
              <div className={`${classes["service-item"]}  position-relative`}>
                <div className={classes["icon"]}>
                  <i className="fa-solid fa-helmet-safety"></i>
                </div>
                <h3>Customer Obsession</h3>
                <p className="h6">
                We work vigorously to earn and keep our customers trust.
                </p>
               
              </div>
            </div>

            <div
              className="col-lg-4 col-md-6 aos-init aos-animate"
              data-aos="fade-up"
              data-aos-delay="600"
            >
              <div className={`${classes["service-item"]}  position-relative`}>
                <div className={classes["icon"]}>
                  <i className="fa-solid fa-arrow-up-from-ground-water"></i>
                </div>
                <h3>Think Big</h3>
                <p className="h6">
                  We constantly raise our bars in every aspect.
                  Thinking small is a self-fulfilling prophecy
                </p>
                
              </div>
            </div>
          </div>
        </div>
      </section>

      <section
        className={`${classes["contact"]} aos-init aos-animate`}
        data-aos="fade-up"
        data-aos-delay="400"
      >
        <div className={classes["overlay"]}>
          <div className={classes["container"]}>
            <div className={classes["row"]}>
              <div
                className="col-md-6 wow "
                data-wow-delay="0.6s"
                style={{
                  visibility: "visible",
                  animationDelay: "0.6s",
                  animationName: "fadeInUp",
                }}
              >
                <div className={classes["fadeInUp"]}>
                  <h2 className="text-uppercase mb-4">Contact Us</h2>
                  <p>
                    Got any sugesstions , requests , queries? Reach us out at
                    below address{" "}
                  </p>
                  <address>
                    <p>
                      <span className={classes["fa"]}>
                        <i className="fa fa-map-marker"></i>
                      </span>
                      Hyderabad, Telangana, India
                    </p>
                    <p>
                      <span className={classes["fa"]}>
                        <i className="fa fa-phone"></i>
                      </span>{" "}
                      0234 234234
                    </p>
                    <p>
                      <span className={classes["fa"]}>
                        <i className="fa fa-envelope"></i>
                      </span>{" "}
                      app@site.com
                    </p>
                  </address>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
      <footer>
        <div className="container">
          <div className="row">
            <p>
              Copyright Â© 2023 Bhanu | Design: <Link to="/">Home</Link>
            </p>
          </div>
        </div>
      </footer>
    </>
  );
};
export default Home;
