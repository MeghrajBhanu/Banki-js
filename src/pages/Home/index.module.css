
.first_container{
    display:flex;
    justify-content: space-around;
}
.image_flex{
    flex:1;
    width:50%;
    margin-top:auto;
    animation: drop 1s ease-in;
}
.content{
    
    flex:1;
    margin-left:0px;
    margin-right:200px;
    justify-content: center;
    margin-top: 100px;
    animation: drop2 1s ease-in;
    
}

.services_card{
    margin-top:0px;
    height:70vh;
    display:flex;
    flex-direction:row;
    
    flex-wrap:wrap;
    justify-content: space-around;
    
}
.title img{
    width:100px;
    margin-left:120px;
    margin-top:40px;
    margin-bottom: 40px;
}

.card-content{
    font-size:20px;
    margin-left:80px;
    font-weight:bolder;
    font-family:'Hello Pretty', sans-serif;
    color: rgb(61, 47, 71);
}
.first_card{
    
    height:300px;
    width:350px;
    background-color: rgb(241, 233, 202);
    margin-top:40px;
    
}
.text{
    
    font-size:40px;
    text-align: center;
    margin-left:290px;
    font-weight:bolder;
    font-family: 'Hello Pretty', sans-serif;
    color: rgb(61, 47, 71);
}
small{

    margin-left:50px;
    font-weight:lighter;
    font-family:'Hello Pretty', sans-serif;
    color: rgb(61, 47, 71);
}
.first_card:hover{
    animation: drop3 1s ease-in;
    
}
.fa{
    display: inline-block;
    background: #28a7e9;
    border-radius: 50%;
    width: 40px;
    height: 40px;
    line-height: 40px;
    text-align: center;
    margin-top: 6px;
    margin-right: 10px;
}
footer {
    background: #fff;
    font-weight: 400;
    text-align: center;
    padding: 20px;
    display: block;
}
.contact {
    background: url('https://templatemo.com/templates/templatemo_446_boxer/images/contact-bg.jpg') no-repeat;
    background-size: cover;
    color: #fff;
    display:block;
}
.container {
    
    margin-left: 100px;
}
.container::before {
    display: table;
    content: " ";
}
.row {
    margin-right: -15px;
    margin-left: -15px;
}
.row:before {
    display: table;
    content: " ";
}
.overlay {
    width: 100%;
    height: 100%;
    background: rgba(0, 0, 0, 0.4);
    padding-top: 90px;
    padding-bottom: 90px;
}
.fadeInUp {
    -webkit-animation-name: fadeInUp;
    animation: fadeInUp 1s ease-in;
    animation-fill-mode: both;
}
.animated {
    -webkit-animation-duration: 1s;
    animation-duration: 1s;
    -webkit-animation-fill-mode: both;
    animation-fill-mode: both;
}
@keyframes fadeInUp {
    0% {
        opacity: 0;
        -webkit-transform: translateY(20px);
        -ms-transform: translateY(20px);
        transform: translateY(20px);
    }
    100% {
        opacity: 1;
        -webkit-transform: translateY(0);
        -ms-transform: translateY(0);
        transform: translateY(0);
    }
}




@media only screen and (max-width: 1024px) {
    .first_container{
        flex-direction:column;
    }

}
@keyframes big {
    0%{
        height:400px;
        width:400px;
    }
    
    100%{
        height:300px;
        width:350px;
    }
}
@keyframes drop {
    0%{
        opacity:0;
        transform:translateX(-80px);
    }
    100%{
        opacity:1;
        transform:translateX(0px);
    }
}
@keyframes drop2 {
    0%{
        opacity:0;
        transform:translateX(80px);
    }
    100%{
        opacity:1;
        transform:translateX(0px);
    }
}
@keyframes drop3 {
    0%{
        opacity:0.5;
        transform:translateY(80px);
    }
    100%{
        opacity:1;
        transform:translateY(0px);
    }
}