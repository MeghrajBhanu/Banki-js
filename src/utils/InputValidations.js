export const validateEmail = (checkingText) => {
    
    const regexp = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/; 
   
    
    if (regexp.exec(checkingText) !== null) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Email Address'
            };
        }
}
export const validatePan = (checkingText) => {
    
    
    if (checkingText.length === 10) {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Pan Number'
            };
        }
}
export const validatePass = (checkingText) => {
    
    
    if (checkingText!=="") {
            return {
                isInputValid: true,
                errorMessage: ''
            };
        } else {
            return {
                isInputValid: false,
                errorMessage: 'Invalid Password'
            };
        }
}