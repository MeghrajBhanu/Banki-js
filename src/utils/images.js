import bank1 from "../assets/bank1.jpg";
import bank2 from "../assets/bank2.png";
import bank3 from "../assets/bank3.jfif";
import bank4 from "../assets/bank4.png";

const image_map={
    "Bank1":bank1,
    "Bank2":bank2,
    "Bank3":bank3,
    "Bank4":bank4,
}
export default image_map;