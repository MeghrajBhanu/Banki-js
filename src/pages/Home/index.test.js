import { render, screen } from "@testing-library/react";
import { onlywrap, renderComponent } from "../../utils/test_helper";
import renderer from 'react-test-renderer';

import Home from ".";

describe("Home Page", () => {
  test("should show Home heading", () => {
    renderComponent(<Home />);
    const value = screen.getAllByText(
      /We are a services group focused on providing you/i
    );
    expect(value[0]).toBeInTheDocument();
  });
  test("snapshot test",()=>{
    const tree = renderer.create(onlywrap(<Home />)).toJSON();
    expect(tree).toMatchSnapshot();
  })
});
