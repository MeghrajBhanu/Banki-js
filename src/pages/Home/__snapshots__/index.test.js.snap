// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page snapshot test 1`] = `
Array [
  <div
    className="first_container"
  >
    <img
      alt="landing"
      className="image_flex"
      src="home.jpg"
    />
    <div
      className="content"
    >
      <p
        className="lead-text text-center p-2 h5 mt-5"
      >
        We are a services group focused on providing you
        <br />
        <br />
        with all your account details associated with different
        <br />
        <br />
        banks at a single point
      </p>
      <h6
        className="text-center p-1"
      >
        Don't have a account yet? 
        <a
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </h6>
    </div>
  </div>,
  <section
    className="services section-bg mt-4"
    id="services"
  >
    <div
      className="container aos-init aos-animate"
      data-aos="fade-up"
    >
      <div
        className="section-header"
      >
        <h2
          style={
            Object {
              "padding": "20px",
            }
          }
        >
          About Us
        </h2>
        <p>
          12Lakh+ customers use our website on daily basis.
        </p>
      </div>
      <div
        className="row gy-4"
      >
        <div
          className="col-lg-4 col-md-6 aos-init aos-animate"
          data-aos="fade-up"
          data-aos-delay="100"
        >
          <div
            className="service-item  position-relative"
          >
            <div
              className="icon"
            >
              <i
                className="fa-solid fa-mountain-city"
              />
            </div>
            <h3>
              Centralizes Access
            </h3>
            <p
              className="h6"
            >
              We provide centralized access to all your bank accounts
            </p>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6 aos-init aos-animate"
          data-aos="fade-up"
          data-aos-delay="200"
        >
          <div
            className="service-item  position-relative"
          >
            <div
              className="icon"
            >
              <i
                className="fa-solid fa-arrow-up-from-ground-water"
              />
            </div>
            <h3>
              Data Security
            </h3>
            <p
              className="h6"
            >
              Keeping your private data safe is our priority.
            </p>
            <a
              className="readmore stretched-link"
              href="service-details.html"
            >
              Learn more 
              <i
                className="bi bi-arrow-right"
              />
            </a>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6 aos-init aos-animate"
          data-aos="fade-up"
          data-aos-delay="300"
        >
          <div
            className="service-item  position-relative"
          >
            <div
              className="icon"
            >
              <i
                className="fa-solid fa-compass-drafting"
              />
            </div>
            <h3>
              Leaders of Market
            </h3>
            <p
              className="h6"
            >
              We are a leading centralized service provider.
            </p>
            <a
              className="readmore stretched-link"
              href="service-details.html"
            >
              Learn more 
              <i
                className="bi bi-arrow-right"
              />
            </a>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6 aos-init aos-animate"
          data-aos="fade-up"
          data-aos-delay="400"
        >
          <div
            className="service-item  position-relative"
          >
            <div
              className="icon"
            >
              <i
                className="fa-solid fa-trowel-bricks"
              />
            </div>
            <h3>
              User Friendly
            </h3>
            <p
              className="h6"
            >
              Our User Experience make us apart from all otherservice providers.
            </p>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6 aos-init aos-animate"
          data-aos="fade-up"
          data-aos-delay="500"
        >
          <div
            className="service-item  position-relative"
          >
            <div
              className="icon"
            >
              <i
                className="fa-solid fa-helmet-safety"
              />
            </div>
            <h3>
              Customer Obsession
            </h3>
            <p
              className="h6"
            >
              We work vigorously to earn and keep our customers trust.
            </p>
          </div>
        </div>
        <div
          className="col-lg-4 col-md-6 aos-init aos-animate"
          data-aos="fade-up"
          data-aos-delay="600"
        >
          <div
            className="service-item  position-relative"
          >
            <div
              className="icon"
            >
              <i
                className="fa-solid fa-arrow-up-from-ground-water"
              />
            </div>
            <h3>
              Think Big
            </h3>
            <p
              className="h6"
            >
              We constantly raise our bars in every aspect. Thinking small is a self-fulfilling prophecy
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="contact aos-init aos-animate"
    data-aos="fade-up"
    data-aos-delay="400"
  >
    <div
      className="overlay"
    >
      <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-md-6 wow "
            data-wow-delay="0.6s"
            style={
              Object {
                "animationDelay": "0.6s",
                "animationName": "fadeInUp",
                "visibility": "visible",
              }
            }
          >
            <div
              className="fadeInUp"
            >
              <h2
                className="text-uppercase mb-4"
              >
                Contact Us
              </h2>
              <p>
                Got any sugesstions , requests , queries? Reach us out at below address
                 
              </p>
              <address>
                <p>
                  <span
                    className="fa"
                  >
                    <i
                      className="fa fa-map-marker"
                    />
                  </span>
                  Hyderabad, Telangana, India
                </p>
                <p>
                  <span
                    className="fa"
                  >
                    <i
                      className="fa fa-phone"
                    />
                  </span>
                   
                  0234 234234
                </p>
                <p>
                  <span
                    className="fa"
                  >
                    <i
                      className="fa fa-envelope"
                    />
                  </span>
                   
                  app@site.com
                </p>
              </address>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <footer>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <p>
          Copyright Â© 2023 Bhanu | Design: 
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </p>
      </div>
    </div>
  </footer>,
]
`;
